java/arcs/core/util/Analytics.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.util
+
+import arcs.core.util.Log.formatter
+import arcs.core.util.Log.writer
+
+/**
+ * Analytics entrance for Arcs.
+ *
+ * Allows for pluggable output sinks (see [logger]).
+ */
+object Analytics {
+
+    enum class Event {
+        StartArc, StopArc
+    }
+
+    interface Logger {
+
+        fun logAllocatorEvent(event: Event, arcName: String, arcId: String)
+
+        fun logArcHostEvent(event: Event, hostId: String, arcId: String)
+
+        fun logArcHostNotFoundException(arcHost: String)
+
+        fun logParticleNotFoundException(particleName: String)
+
+    }
+
+    var logger: Logger = DEFAULT_LOGGER
```

Comment:
Provide KDoc, especially around why/when it makes sense to change the value (given that it's a `var`).

Also, this should go at the top of the `object`, with the interfaces/enums below it.